lib/model-builder.js

```diff
@@ -308,7 +308,9 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
   // Add the id property
   if (idInjection) {
     // Set up the id property
-    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true});
+    // updateOnly property is added to indicate that this property will appear in the model for update/updateorcreate
+    // operations but and not for create operation.
+    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true, updateOnly: true});
```

Comment:
> what you are proposing is to move only the forceId determining logic from setUpDataAccess and leave rest of the logic as is.. for e.g setting idType and calling modelClass.validatesAbsenceOf(..) based on
forceId from the settings.
 
> I did try your proposed solution and it does work.

Cool!

I think it probably makes sense to move `modelClass.validatesAbsenceOf` over too, while we are at cleaning up this code. Having said that, what you are proposing in your comment above looks reasonable too.